cli/dcoscli/node/main.py

```diff
@@ -491,32 +489,32 @@ def _log(follow, lines, leader, slave, component, filters):
     :rtype: int
     """
 
-    if not (leader or slave) or (leader and slave):
-        raise DCOSException(
-            'You must choose one of --leader or --mesos-id.')
+    if not (leader or slave):
+        raise DCOSException('You must choose one of --leader or --mesos-id.')
 
     if lines is None:
         lines = 10
 
     lines = util.parse_int(lines)
 
-    try:
-        _dcos_log(follow, lines, leader, slave, component, filters)
+    # if journald logging is disabled. Read from files API and exit.
+    if not log.dcos_log_enabled():
```

Comment:
We want to keep serving `component` logs via the CLI even if the journald logger for mesos is disabled. The only thing the CLI should deprecate when the journald logger for Mesos is disabled is using the journald API to get task logs - for that, it should use the files API (which this PR already provides). CC @spahl